### DISCUSSION TITLE
Skip update hook

### DISCUSSION THREAD

**Discussion by rpradal** [containrrr/watchtower#647] [Q&A]
Hi,

I would like to be able to tell watchtower to skip an update if my container is in a particular state.
Do you have any idea how i could achieve that ?

Currently, i have a pre-update hook where i wait for my container to be in an updatable state before finishing, but this is far from ideal...

Thanks !

**Comment by zoispag** [comment]
Hi there,

I cannot think of a way different right now than the pre-update hook. Maybe trying to find a working solution for your scenario, could you please share some more information? What kind of state checking mechanism do you have in mind? What is the desired state before allowing an update to happen?

**Reply by rpradal** [reply]
My use case is the following : 

I am building a kiosk app divided in several containers (without persistant storage). Users can login to this kiosk app and perform some operations and eventually log out. I would like to skip any update if the kiosk state is not idle to provide my users an undisrupted experience.

If the `pre-update` exit code handling behaviour would be to skip the update if an exit code is different from 0, that could do the job, but this is not the implemented behaviour...

**Reply by zoispag** [reply]
There is a workaround by setting the timeout to 0 and force it to wait for an exit code. As long as your pre-update hook script is running, the pre-update will not continue. You can use the `com.centurylinklabs.watchtower.lifecycle.pre-update-timeout` and set it to `0`.

**Reply by rpradal** [reply]
Oh yes that is what we do (i forgot to mention it). The drawback is that we have to be super careful when writing our `pre-update` script or we can be stuck in an infinite loop.

**Reply by zoispag** [reply]
Yes, that is indeed a risk that we also discussed when adding this feature. @simskij is it possible for watchtower to read the exit code and stop the update with a non-zero? This would stop the execution right away, without having to add a blocking script without timeout to make it work.

**Reply by simonaronsson** [reply]
Yeah, to be totally honest, I already thought we did this! 😅  This is the intended behavior, so we'll need to fix that. Thanks for reporting!
